### PR TITLE
Update OO controllers code snippets in documentation

### PR DESCRIPTION
**Pull request by stefaniapedrazzi**
Fix https://github.com/omichel/webots/issues/3659: use the same structure as in the template controllers

### CODE REVIEW

**Comment by stefaniapedrazzi** [comment]
:+1:

File: guide/cpp-java-python.md [hunk 1]
```diff
@@ -81,41 +81,25 @@ etc. The various devices instances can be obtained with dedicated methods of the
...
+  Robot *robot = new Robot();
 
-    void run() {
-      // Main control loop
-      while (step(TIME_STEP) != -1) {
-        // Read the sensors
-        double val = distanceSensor->getValue();
+  int timeStep = (int) robot->getBasicTimeStep();
+  LED *led = getLED("ledName");
+  DistanceSensor *distanceSensor = getDistanceSensor("distanceSensorName");
```
Comment (omichel) [inline guide/cpp-java-python.md]: Shouldn't it be robot->getDistanceSensor()?

File: guide/cpp-java-python.md [hunk 2]
```diff
@@ -81,41 +81,25 @@ etc. The various devices instances can be obtained with dedicated methods of the
...
+  distanceSensor->enable(timeStep);
 
-        // Process sensor data here
+  // Main control loop
+  while (step(timeStep) != -1) {
```
Comment (omichel) [inline guide/cpp-java-python.md]: And robot->step(timeStep)?

File: guide/cpp-java-python.md [hunk 3]
```diff
@@ -126,21 +110,19 @@ int main(int argc, char **argv) {
...
-    led = getLED("my_led");
-    distanceSensor = getDistanceSensor("my_distance_sensor");
-    distanceSensor.enable(TIME_STEP);
-  }
+  public static void main(String[] args) {
+    Robot robot = new Robot();
+
+    int timeStep = (int) Math.round(robot.getBasicTimeStep());
+    LED led = getLED("my_led");
+    DistanceSensor distanceSensor = getDistanceSensor("my_distance_sensor");
```
Comment (omichel) [inline guide/cpp-java-python.md]: robot.getDistanceSensor(...)

File: guide/cpp-java-python.md [hunk 4]
```diff
@@ -126,21 +110,19 @@ int main(int argc, char **argv) {
...
+    distanceSensor.enable(timeStep);
 
-  public void run() {
     // main control loop
-    while (step(TIME_STEP) != -1) {
+    while (step(timeStep) != -1) {
```
Comment (omichel) [inline guide/cpp-java-python.md]: robot.step(timeStep)

etc.

**Comment by stefaniapedrazzi** [comment]
:+1:

File: reference/supervisor.md [hunk 5]
```diff
@@ -583,25 +583,22 @@ case the controller should call its destructor. Here is a C++ example:
...
+  while (! finished) {
+    // your controller code here
     ...
-    while (! finished) {
-      // your controller code here
-      ...
-      step(TIME_STEP);
-    }
-    simulationQuit(EXIT_SUCCESS);  // ask Webots to terminate
+    if (step(TIME_STEP) == -1)
```
Comment (omichel) [inline reference/supervisor.md]: should be controller->step(TIME_STEP)

**Comment by stefaniapedrazzi** [comment]
:+1:

**Comment by omichel** [comment]
:+1:

**Pull request merged**